ksql-cli/src/main/java/io/confluent/ksql/cli/Cli.java

```diff
@@ -213,16 +213,17 @@ public void runNonInteractively(final String input) throws Exception {
   }
 
   private List<String> getLogicalLines(final String input) {
-    // TODO: Convert the input string into an InputStream, then feed it to the terminal via
-    // TerminalBuilder.streams(InputStream, OutputStream)
+    // TODO: Convert the input string into an InputStream, then feed it to
+    // the terminal via TerminalBuilder.streams(InputStream, OutputStream)
     final List<String> result = new ArrayList<>();
     StringBuilder logicalLine = new StringBuilder();
     for (final String physicalLine : input.split("\n")) {
-      if (!physicalLine.trim().isEmpty()) {
-        if (physicalLine.endsWith("\\")) {
-          logicalLine.append(physicalLine.substring(0, physicalLine.length() - 1));
+      final String trimmedPhysicalLine = physicalLine.trim();
+      if (!trimmedPhysicalLine.isEmpty()) {
+        if (trimmedPhysicalLine.endsWith("\\")) {
+          logicalLine.append(trimmedPhysicalLine.replaceAll("\\\\", " "));
         } else {
-          result.add(logicalLine.append(physicalLine).toString().trim());
+          result.add(logicalLine.append(trimmedPhysicalLine.replaceAll("\\\\", " ")).toString());
```

Comment:
There should be no need to replace slashes in this line as it does not end with a slash.